compiler/rustc_borrowck/src/polonius/liveness_constraints.rs

```diff
@@ -231,11 +231,9 @@ struct VarianceExtractor<'a, 'tcx> {
 impl<'tcx> VarianceExtractor<'_, 'tcx> {
     fn record_variance(&mut self, region: ty::Region<'tcx>, variance: ty::Variance) {
         // We're only interested in the variance of vars and free regions.
-
-        if region.is_bound() || region.is_erased() {
-            // ignore these
+        let Some(region) = self.universal_regions.try_to_region_vid(region) else {
```

Comment:
Hmm, so this seems interesting. Where would we be seeing a `Bound` or `Erased` region?  Seems curious to me that there's some subset of regions that we're essentially just ignoring.

Reply:
Could be worth adding a comment explaining the cases this might happen.

Reply:
This came about like so: 
- I wanted the variance for vars and free regions and could only get it by duplicating the Generalizer's trick with `TypeRelation`, which unfortunately also yields regions that can't be converted to vids. 
- so I just ignored these ICE causing kinds and saw no particular changes in the tests (though there are still 6 failures left to diagnose and they could hiding in there): as far as I know there were no live regions that lacked variance due to this. There's also similarly [unexpected kinds reaching borrowck](https://github.com/search?q=repo%3Arust-lang%2Frust+cannot+convert+to+a+region+vid&type=issues), due to issues upstream and/or the ongoing effort of continuing compilation of code as deep as possible, that could be involved. To see whether we should be trying to e.g. recurse after skipping or discharging the binder, I casually brought it up with Niko who said "seems fine to ignore for now" and we didn't think more of it since then. 

Since then I also discovered our test coverage is actually very weak in that area, so I'll ask you what is the correct thing to do here?

Reply:
Personally, I would prefer to just fail to compile if we ever hit this (as we do for the examples you linked).

Really, I think there's an underlying problem here that we should investigate.

Reply:
Sure, I'll put the ICEs back in then, and we can recheck and investigate later.

Reply:
Yeah, for sure we can always come back to it.

Reply:
> Sure, I'll put the ICEs back in then, and we can recheck and investigate later.

Actually maybe not, I forgot there's too many of them and I don't believe they really matter here: we're not ignoring regions in the graph because of the (transparent to reachability) fallback if no variance is found for a region.  As for `'erased` though, it was in [this test]() that involves GCE amongst others, so imho it's fine to ignore for now.

A fixme like you initially suggested looks best under these circumstances. I'll ping you on zulip to discuss.

Reply:
A FIXME is fine.

Reply:
Ok cool, I've pushed a fixme in the last commit, explaining why it's fine to not have variance data for these for now but we still need to look into it.